src/test/java/teammates/ui/webapi/FeedbackSessionLogsUpdateActionTest.java

```diff
@@ -0,0 +1,130 @@
+package teammates.ui.webapi;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.FeedbackSessionLogEntryAttributes;
+import teammates.common.datatransfer.logs.FeedbackSessionLogType;
+import teammates.common.util.Const;
+import teammates.test.ThreadHelper;
+
+/**
+ * SUT: {@link FeedbackSessionLogsUpdateAction}.
+ */
+public class FeedbackSessionLogsUpdateActionTest
+        extends BaseActionTest<FeedbackSessionLogsUpdateAction> {
+
+    private static final int MAX_WINDOW_SIZE = 1;
+
+    @Override
+    protected String getActionUri() {
+        return Const.CronJobURIs.AUTOMATED_FEEDBACK_SESSION_LOGS_UPDATE;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @Override
+    @Test
+    protected void testAccessControl() {
+        // verifyOnlyAdminCanAccess();
+    }
+
+    @Override
+    @Test
+    public void testExecute() throws Exception {
+        FeedbackSessionLogsUpdateAction action = getAction();
+
+        ______TS("creating 2 logs in the same window should return only 1 log");
+
+        // create a window exceeding max size
+        for (int i = 0; i < MAX_WINDOW_SIZE + 1; i++) {
+            mockLogsProcessor.createFeedbackSessionLog("idOfTypicalCourse1",
+                    "student1InCourse1@gmail.tmt", "First feedback session", FeedbackSessionLogType.ACCESS.getLabel());
+        }
+
+        action.execute();
+
+        Instant currentTime = Instant.now();
+        List<FeedbackSessionLogEntryAttributes> logs = logic.getFeedbackSessionLogs(
+                "idOfTypicalCourse1", null,
+                currentTime.minus(1, ChronoUnit.SECONDS).toEpochMilli(),
+                currentTime.toEpochMilli(), null);
+
+        assertEquals(1, logs.size());
+
+        FeedbackSessionLogEntryAttributes log = logs.get(0);
+
+        assertEquals("student1InCourse1@gmail.tmt", log.getStudentEmail());
+        assertEquals("First feedback session", log.getFeedbackSessionName());
+        assertEquals(FeedbackSessionLogType.ACCESS.getLabel(), log.getFeedbackSessionLogType());
+
+        // wait 5s to isolate the previous log window
+        ThreadHelper.waitFor(5 * 1000);
```

Comment:
Every place where `5 * 1000` and `2 * 1000` are defined in this class are magic numbers. They are based on a certain field in the class being tested.